src/cargo/ops/cargo_new.rs

```diff
@@ -141,28 +297,90 @@ fn mk(config: &Config, path: &Path, name: &str,
         (Some(name), None, _, None) |
         (None, None, name, None) => name,
     };
+    
+    let mut cargotoml_path_specifier = String::new();
+    
+    let mut there_was_already_a_lib = false;
+    let mut previous_lib_relpath = String::new();
+    
+    // Calculare what [lib] and [[bin]]s do we need to append to Cargo.toml
+    
+    let explicit_better_than_implicit = opts.source_files.len() > 1;
```

Comment:
It should be ok to omit this, Cargo will infer `src/lib.rs` or `src/main.rs` if left out, so if those files are used then it can just be ignored.